### DISCUSSION TITLE
How to add lang attribute to the html tag without using custom `_app.tsx` or `_document.tsx` files.

### DISCUSSION THREAD

**Discussion by justin-v-ds** [vercel/next.js#50352] [Help]
### Summary

On my next.js app, whenever I click on `view page code` on the browser, it appears that `html` tag doesn't have `lang` attribute by default. At first time, I've tried the following solution modifying `_document.tsx` file.

```
import NextDocument, { Html, Head, Main, NextScript } from 'next/document';
import type { DocumentContext, DocumentInitialProps } from 'next/document';

function Document({
  query,
}: DocumentInitialProps & Pick<DocumentContext, 'query'>) {
  const lang = query.page ? query.page[0].split('-')[0] : 'en';

  return (
    <Html lang={lang}>
      <Head />
      <body>
        <Main />
        <NextScript />
      </body>
    </Html>
  );
}

Document.getInitialProps = async (context: DocumentContext) => {
  const initialProps = await NextDocument.getInitialProps(context);
  return { ...initialProps, query: context.query };
};

export default Document;
```

But after this change, I've noticed that it causes pages to no longer generate statically. So for now, I wanna approach adding lang attribute without using custom `_app.tsx` or `_document.tsx` files. The important thing here is the `lang` attribute should be displayed when user clicks `view page source` vs `inspect elements`. Happy to discuss any thoughts or suggestions. thanks!

**Comment by bc-pool** [comment]
@justin-v-ds  did you solve it?
